rfcs/2023-05-03-data-volume-metrics.md

```diff
@@ -0,0 +1,146 @@
+# RFC 2023-05-02 - Data Volume Insights metrics
+
+Vector needs to be able to emit accurate metrics that can be usefully queried
+to give users insights into the volume of data moving through the system.
+
+## Scope
+
+### In scope
+
+- All event metrics within Vector need to emit the estimated JSON size of the
+  event. With a consistent method for determining the size it will be easier
+  to accurately compare data in vs data out.
+- The metrics sent by each sink needs to be tagged with the source id of the
+  event so the route an event takes through Vector can be queried.
+- Each event needs to be labelled with a `service`. This is a new concept
+  within Vector and represents the application that generated the log or
+  metric.
+- The service tag and source tag in the metrics needs to be opt in so customers
+  that don't need the increased cardinality are unaffected.
+
+### Out of scope
+
+- A separate metric, `component_sent_bytes_total` that indicates network bytes
+  sent by Vector is not considered here.
+
+## Pain
+
+Currently it is difficult to accurately gauge the volume of data that is moving
+through Vector. It is difficult to query where data being sent out has come
+from.
+
+## Proposal
```

Comment:
Could we list out the new timeseries/tags that we be published? I think that'll help reviewers understand this change.

Reply:
I'm not sure I understand what you mean? Do you mean a complete list of all the metric names together with their associated  tags that Vector publishes?

Reply:
> Could we list out the new timeseries/tags that we be published? I think that'll help reviewers understand this change.

I'm not sure I understand what you mean? Do you mean a complete list of all the metric names together with their associated tags that Vector publishes?

Reply:
Ah, yeah, basically that. I think that would help readers understand exactly what is changing. For example something like:

```
component_received_event_bytes_total{component_id, component_name, component_type}
```

changing to

```
component_received_event_bytes_total{component_id, component_name, component_type, service, source_id}
```

For each of the metrics that will be updated here.